src/OrbitGl/TimeGraph.cpp

```diff
@@ -634,12 +636,24 @@ void TimeGraph::DrawAllElements(Batcher& batcher, TextRenderer& text_renderer,
 
   DrawContext context{current_mouse_time_ns, picking_mode};
   Draw(batcher, text_renderer, context);
+  DrawChildrenSeparator(batcher);
 
   if ((!picking && update_primitives_requested_) || picking) {
     PrepareBatcherAndUpdatePrimitives(picking_mode);
   }
 }
 
+void TimeGraph::DrawChildrenSeparator(Batcher& batcher) {
```

Comment:
Do I understand correctly that the separator is a box with background color that's drawn on top of other elements? I don't like this, I can see this creating issues in the future because it's not what you'd expect when looking at the UI. Think of the rounded corners, we're having issues there because of the "hacky" way we're using the background color.

What's the reason to do it like this? Can't you just leave it "blank" instead of drawing a separator at all? I assume that then certain elements (iterators?) would be drawn inside the separators, but that should be fixed by correctly parenting and positioning the iterators.

Reply:
The reason is 100% visual. It was discussed with Carsten, but maybe we can rediscuss it today.

It's a hack, and I should add a TODO here, you are right. It's a hack because CaptureViewElements are not forcing to only draw in their rectangle, so without this, TrackContainer is drawing outside in this margin anyway.

Reply:
Iterators are working perfectly. The problem are the Tracks.

Reply:
Ok, got it. But then I maybe wouldn't generalize it to all children, but just add the spacing and the separator after the track container.

You *could* create a new CaptureViewElement `Separator` and place that in between the container and the timeline... not sure if this is overkill. In any case, I think you should add a TODO here and we need to discuss if it's worth the effort to restrict elements to their parents (which it probably isn't...).

Reply:
These are 2 different topics:
- Generalizing the separator or not. I started with a hack (only that separator) and I didn't like it so I decided to generalize it, but I agree that could have been a mistake. Maybe the best option is consider this separator as a hack (which it is) with the TODO there.

- CaptureViewElement or not. I think that the ideal solution is restricting elements to only draw in their space. I wouldn't go for a slightly more elegant solution if we don't want to go there. Maybe a quick-sync here?